File: dashboard/src/utils/UnitConverter.ts
export default class UnitConverter {
    static bytesToHumanReadable(bytes: number): { value: number; label: string } {
        const GIGABYTE = 1024 * 1024 * 1024;
        const MEGABYTE = 1024 * 1024;

        if (bytes >= GIGABYTE) {
            return {value: parseFloat((bytes / GIGABYTE).toFixed(2)), label: 'GB'};
        } else if (bytes >= MEGABYTE) {
            return {value: parseFloat((bytes / MEGABYTE).toFixed(2)), label: 'MB'};
        } else {
            return {value: parseFloat((bytes / 1024).toFixed(2)), label: 'KB'};
        }
    }

    static bitsToHumanReadable(bits: number): { value: number; label: string } {
        const GIGABIT = 1_000_000_000;
        const MEGABIT = 1_000_000;
        const KILOBIT = 1_000;

        if (bits >= GIGABIT) {
            return {value: parseFloat((bits / GIGABIT).toFixed(2)), label: 'Gbps'};
        } else if (bits >= MEGABIT) {
            return {value: parseFloat((bits / MEGABIT).toFixed(2)), label: 'Mbps'};
        } else if (bits >= KILOBIT) {
            return {value: parseFloat((bits / KILOBIT).toFixed(2)), label: 'Kbps'};
        } else {
            return {value: bits, label: 'bps'};
        }
    }
}

File: system_monitor/src/unit_converter.rs
pub struct UnitConverter;

impl UnitConverter {
    pub fn bytes_to_human_readable(bytes: u64) -> String {
        const GIGABYTE: u64 = 1024 * 1024 * 1024;
        const MEGABYTE: u64 = 1024 * 1024;

        if bytes >= GIGABYTE {
            format!("{:.2} GB", bytes as f64 / GIGABYTE as f64)
        } else if bytes >= MEGABYTE {
            format!("{:.2} MB", bytes as f64 / MEGABYTE as f64)
        } else {
            format!("{:.2} KB", bytes as f64 / 1024.0)
        }
    }
    pub fn bits_to_human_readable(bits: u64) -> String {
        const GIGABIT: u64 = 1_000_000_000;
        const MEGABIT: u64 = 1_000_000;
        const KILOBIT: u64 = 1_000;

        if bits >= GIGABIT {
            format!("{:.2} Gbps", bits as f64 / GIGABIT as f64)
        } else if bits >= MEGABIT {
            format!("{:.2} Mbps", bits as f64 / MEGABIT as f64)
        } else if bits >= KILOBIT {
            format!("{:.2} Kbps", bits as f64 / KILOBIT as f64)
        } else {
            format!("{} bps", bits)
        }
    }
}


